au/code/au/quantity.hh

```diff
@@ -424,6 +424,22 @@ class Quantity {
     Rep value_{};
 };
 
+// Give more readable error messages when passing `Quantity` to a unit slot.
+template <typename U, typename R>
+struct AssociatedUnit<Quantity<U, R>> {
+    static_assert(
+        detail::AlwaysFalse<U, R>::value,
+        "Can't pass `Quantity` to a unit slot (see: "
+        "https://aurora-opensource.github.io/au/main/troubleshooting/#quantity-to-unit-slot)");
```

Comment:
Do we have any way to ensure that the link remains valid?

Reply:
Good question!

The simple answer is "not directly".  But one thing we _can_ do is to add this same link _somewhere else_... or at least, the relative parts of the URL.  This will mean that if we ever change the anchor name, then `au-docs-serve` will throw up errors like the following, and we're likely to notice it at some point and fix the docs.

![image](https://github.com/user-attachments/assets/8d35a5e5-be83-4a6b-9da5-45cd0cf65a16)

Incidentally... this screenshot shows actual doc errors that I had introduced in this PR.  :flushed:  So, I fixed those too.

And then I added a new section to the "unit slot" docs which points back to the troubleshooting guide, which feels like it "should have been there all along", and --- more to the point here --- which also uses the URL from these `static_assert` calls, giving it the best _partial_ coverage that I know how to get.

I'm not planning to ask for a re-review, but if you spot anything we should fix, we can always do a quick follow on PR.